### PR TITLE
Fix Typescript types to align with docs and support EntityConfiguration.indentifiers option

### PR DESCRIPTION
**Pull request by Max101**
Hi, apologies but there does not seem to be a right way to suggest a fix in this repo or otherwise I missed it.

There is an option to override the default identifiers but it is not available in the typescript config. Also, the `composite` key in the index configuration is optional when one uses the `template`. 

TLDR: I am migrating an existing OneTable project to electroDb and I have to support things done slightly differently. None-the-less the types provided in this file right now are wrong

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -2232,6 +2232,10 @@ export type EntityConfiguration = {
     client?: DocumentClient;
     listeners?: Array<ElectroEventListener>;
     logger?: ElectroEventListener;
+    identifiers?: {
```
Comment (tywalch) [inline index.d.ts:2235]: This is a great add!

File: index.d.ts [hunk 2]
```diff
@@ -1532,13 +1532,13 @@ export interface Schema<A extends string, F extends string, C extends string> {
             readonly pk: {
                 readonly casing?: "upper" | "lower" | "none" | "default";
                 readonly field: string;
-                readonly composite: ReadonlyArray<F>;
+                readonly composite?: ReadonlyArray<F>;
```
Comment (tywalch) [inline index.d.ts:1535]: Hey @Max101 👋

While this is supported by the library, the type-system depends on this composite being present which is why there is a little disconnect here between the types and the validation.

**Review by tywalch — CHANGES_REQUESTED** [review]
The `composite` attribute will need to remain required for now

File: index.d.ts (hunk 2 above)
Comment (Max101) [reply index.d.ts:1553]: right ok I understand. We have defaulted to an empty array for now

**Review by tywalch — APPROVED** [review]

**Pull request merged**